internal/events/event_manager.go

```diff
@@ -168,20 +175,34 @@ func (em *eventManager) CreateDurableSubscription(ctx context.Context, subDef *f
 	// Do a check first for existence, to give a nice 409 if we find one
 	existing, _ := em.database.GetSubscriptionByName(ctx, subDef.Namespace, subDef.Name)
 	if existing != nil {
-		return i18n.NewError(ctx, i18n.MsgAlreadyExists, "subscription", subDef.Namespace, subDef.Name)
-	}
-
-	// We lock in the starting sequence at creation time, rather than when the first dispatcher
-	// starts, as that's a more obvious behavior for users
-	sequence, err := calcFirstOffset(ctx, em.database, subDef.Options.FirstEvent)
-	if err != nil {
-		return err
+		if mustNew {
+			return i18n.NewError(ctx, i18n.MsgAlreadyExists, "subscription", subDef.Namespace, subDef.Name)
+		}
+		// Copy over the generated fields, so we can do a compare
+		subDef.Created = existing.Created
+		subDef.ID = existing.ID
+		subDef.Updated = fftypes.Now()
+		subDef.Options.FirstEvent = existing.Options.FirstEvent // we do not reset the sub position
+		existing.Updated = subDef.Updated
+		def1, _ := json.Marshal(existing)
+		def2, _ := json.Marshal(subDef)
+		if bytes.Equal(def1, def2) {
+			log.L(ctx).Infof("Subscription already exists, and is identical")
+			return nil
+		}
```

Comment:
Just out of curiosity wondering if this is the best way to compare the subscriptions, by serializing them to JSON and comparing the bytes?

Reply:
Alternatives I thought through, all seemed to have the characteristic that if the structure changed, then this code would need to be updated to know about new fields.
As subscription options are pluggable by implementations, that's a lot of scope for the code to stop doing its job.

Any ideas for another way of comparing these nested structures (noting multiple pointers in them)?

Reply:
Those are good points. Hand't considered the plugin factor. My first thought would be to just have a `compare` method on the struct itself. But you are correct that if the structure changes then the comparison function would also have to change.